### PR TITLE
add sql state and vendorCode

### LINKED ISSUE #4654: Throw correct SQLState and vendorCode with SQLException when out of index from ResultSetMetaData.getColumnName()
The current ResultSetMetaData can throw SQLException without SQL State. Please fill the correct SQLState and vendorCode in SQLException.

Please fix on classes under `TODO fill correct SQL state` comment:

- [ ] ShardingResultSetMetaData
- [ ] EncryptResultSetMetaData

### PR DESCRIPTION
**Pull request by zjcnb**
Fixes #4654

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/ShardingResultSetMetaData.java [hunk 1]
```diff
@@ -35,65 +36,64 @@
  */
 @RequiredArgsConstructor
 public final class ShardingResultSetMetaData extends WrapperAdapter implements ResultSetMetaData {
-    
+
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/ShardingResultSetMetaData.java]: Could you keep the original indent?

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java [hunk 2]
```diff
@@ -25,6 +25,10 @@
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class SQLExceptionConstant {
-    
+
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java]: Could you keep the original indent?

**Comment by terrymanu** [comment]
The commit log is not make sense, could you reference the contributor guide to make a clear develop flow of commit?
FIY: https://shardingsphere.apache.org/community/en/contribute/contributor/

**Comment by zjcnb** [comment]
> Can you change the title to a whole sentence for search friendly?

ok. i done

**Comment by terrymanu** [comment]
Because the commit log is not clean for this pr. I just colse it, please submit a new pr with clean commit log.

**Pull request closed without merging**